From 3f02f3b6befd6c4a613c7ecc8924c4662ffcdd4a Mon Sep 17 00:00:00 2001
From: Anton Staroverov <73071319+tonystar@users.noreply.github.com>
Date: Fri, 27 Mar 2026 16:29:39 -0600
Subject: [PATCH] import base config from Blades

---
 eleventy.config.js | 114 ++++-----------------------------------------
 package.json       |   2 +-
 2 files changed, 11 insertions(+), 105 deletions(-)

diff --git a/eleventy.config.js b/eleventy.config.js
index 79ad385..36366ad 100644
--- a/eleventy.config.js
+++ b/eleventy.config.js
@@ -1,42 +1,9 @@
-/* Plugins */
-import { RenderPlugin } from "@11ty/eleventy";
-import eleventyBladesPlugin from "@anydigital/eleventy-blades";
-/* Dynamic plugins */
-let eleventyNavigationPlugin;
-try {
-  eleventyNavigationPlugin = (await import("@11ty/eleventy-navigation")).default;
-} catch (e) {
-  // @11ty/eleventy-navigation not installed
-}
-let pluginTOC;
-try {
-  pluginTOC = (await import("@uncenter/eleventy-plugin-toc")).default;
-} catch (e) {
-  // @uncenter/eleventy-plugin-toc not installed
-}
+/* Base Eleventy Blades config */
+import baseConfig from "@anydigital/eleventy-blades/base-config";
+
 /* Libraries */
 import markdownIt from "markdown-it";
-/* Dynamic libraries */
-let slugify;
-try {
-  slugify = (await import("@sindresorhus/slugify")).default;
-} catch (e) {
-  // @sindresorhus/slugify not installed
-}
-let markdownItAnchor;
-try {
-  markdownItAnchor = (await import("markdown-it-anchor")).default;
-} catch (e) {
-  // markdown-it-anchor not installed
-}
-let markdownItAttrs;
-try {
-  markdownItAttrs = (await import("markdown-it-attrs")).default;
-} catch (e) {
-  // markdown-it-attrs not installed
-}
-/* Data */
-import yaml from "js-yaml";
+import markdownItAnchor from "markdown-it-anchor";
 
 /* Minform */
 import minformConfig from "./minform.config.js";
@@ -47,76 +14,15 @@ import minformConfig from "./minform.config.js";
  * @returns {Object} The Eleventy configuration object
  */
 export default function (eleventyConfig) {
-  const inputDir = eleventyConfig.directories.input;
-
-  /* Jekyll parity */
-  eleventyConfig.addPassthroughCopy("assets");
-  eleventyConfig.addGlobalData("layout", "default");
-  eleventyConfig.setLiquidOptions({ dynamicPartials: true, jsTruthy: true });
-  eleventyConfig.addFilter("relative_url", (content) => content);
-
-  /* Plugins */
-  eleventyConfig.addPlugin(RenderPlugin);
-  if (eleventyNavigationPlugin) eleventyConfig.addPlugin(eleventyNavigationPlugin);
-  eleventyConfig.addPlugin(eleventyBladesPlugin, {
-    mdAutoNl2br: true,
-    mdAutoRawTags: false,
-    autoLinkFavicons: true,
-    siteData: true,
-    filters: [
-      "attr_set",
-      "attr_includes",
-      "merge",
-      "remove_tag",
-      "if",
-      "attr_concat",
-      "fetch",
-      "section",
-      "strip_tag",
-      "unindent",
-    ],
-  });
-  if (pluginTOC) {
-    eleventyConfig.addPlugin(pluginTOC, {
-      ignoredElements: ["sub", ".header-anchor"],
-      ul: true,
-    });
-  }
+  baseConfig(eleventyConfig);
 
-  /* Libraries */
-  let md = markdownIt({
-    html: true,
-    linkify: true,
-  });
-  if (markdownItAnchor) {
-    md = md.use(markdownItAnchor, {
-      slugify: slugify, // @TODO: TRICKS
-      permalink: markdownItAnchor.permalink.ariaHidden({
-        class: null,
-        renderAttrs: () => ({ "data-is-anchor": true }),
-      }),
-    });
+  // Override defaults:
+  const bladesPlugin = eleventyConfig.plugins.find((p) => p.plugin.name === "eleventyBladesPlugin");
+  if (bladesPlugin) {
+    bladesPlugin.options.mdAutoRawTags = false;
   }
-  if (markdownItAttrs) md = md.use(markdownItAttrs);
-  // eleventyConfig.setLibrary("md", md);
+  eleventyConfig.setLiquidOptions({ dynamicPartials: true, jsTruthy: true });
   eleventyConfig.setLibrary("md", markdownIt().use(markdownItAnchor));
-  eleventyConfig.addFilter("markdownify", (content) => md.render(String(content ?? "")));
-
-  /* Data */
-  eleventyConfig.addDataExtension("yml", (contents) => yaml.load(contents));
-
-  /* Build */
-  eleventyConfig.addPassthroughCopy(
-    {
-      _public: ".",
-      ...(inputDir !== "." && { [`${inputDir}/_public`]: "." }),
-    },
-    { expand: true }, // This follows/resolves symbolic links
-  );
-
-  /* Dev tools */
-  // Follow symlinks in Chokidar used by 11ty to watch files
-  eleventyConfig.setChokidarConfig({ followSymlinks: true });
 
   minformConfig(eleventyConfig);
 }
diff --git a/package.json b/package.json
index e68f095..8d486d4 100644
--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
     "@11ty/eleventy-navigation": "^1.0.5",
     "@anydigital/blades": "^0.27.0-alpha.14",
     "@anydigital/breakout-css": "^0.11.1",
-    "@anydigital/eleventy-blades": "^0.28.0-alpha.7",
+    "@anydigital/eleventy-blades": "^0.28.0-alpha.14",
     "@picocss/pico": "^2.1.1",
     "@tailwindcss/cli": "^4.2.1",
     "markdown-it": "^14.1.0",
-- 
2.47.3

